extensions/map-layers/src/ui/widget/MapLayerDroppable.tsx

```diff
@@ -68,12 +69,13 @@ export function MapLayerDroppable(props: MapLayerDroppableProps) {
               const indexInDisplayStyle = props.activeViewport?.displayStyle.findMapLayerIndexByNameAndUrl(activeLayer.name, activeLayer.url, activeLayer.isOverlay);
               if (indexInDisplayStyle !== undefined && indexInDisplayStyle >= 0) {
                 const layerSettings = props.activeViewport.displayStyle.mapLayerAtIndex(indexInDisplayStyle, activeLayer.isOverlay);
-
-                ModalDialogManager.openDialog(<MapUrlDialog activeViewport={props.activeViewport}
-                  isOverlay={props.isOverlay}
-                  layerRequiringCredentials={layerSettings?.toJSON()}
-                  onOkResult={props.onItemEdited}
-                  mapTypesOptions={props.mapTypesOptions}></MapUrlDialog>);
+                if (layerSettings instanceof ImageMapLayerSettings) {
```

Comment:
Assuming call to findMapLayerIndexByNameAndUrl() for a model maplayer will always return -1 , this check is probably never going to hit.  Also, we might want to call findMapLayerIndexByNameAndUrl earlier, and skip the rendering of button in case of model map layer, because anyway it will never requiere auth.

Reply:
Fixed --- findMapLayerIndexByNameAndSource will return ModelMapLayerSettings